src/plugins/grpc/client.js

```diff
@@ -0,0 +1,100 @@
+'use strict'
+
+const { FORMAT_HTTP_HEADERS, Tags } = require('opentracing')
+
+function patch (grpc, tracer, config) {
+  if (config.client === false) return
+
+  config = config.client || config
+
+  function gRPCInterceptor (options, nextCall) {
+    const { path, responseStream } = options.method_definition
+    const scope = tracer.scopeManager().active()
+    const span = tracer.startSpan('grpc.client', {
+      childOf: scope && scope.span(),
+      tags: {
+        [Tags.SPAN_KIND]: Tags.SPAN_KIND_RPC_CLIENT,
+        'grpc.method_name': path,
+        'grpc.method_type': responseStream ? 'server_stream' : 'unary',
+        'resource.name': path.split('/').slice(-1),
+        'service.name': config.service || `${tracer._service}-grpc`
+      }
+    })
+
+    return new grpc.InterceptingCall(nextCall(options), {
+      sendMessage: (message, next) => {
+        if (config.arguments) {
+          const values = config.arguments(message)
+
+          for (const key in values) {
+            span.setTag(`grpc.arguments.${key}`, values[key])
+          }
+        } else {
+          span.setTag(`grpc.arguments`, message)
+        }
+
+        next(message)
+      },
+
+      start: (metadata, _listener, next) => {
+        if (metadata) {
+          for (const [key, value] of Object.entries(metadata.getMap())) {
+            span.setTag(`grpc.metadata.${key}`, value)
+          }
+        }
+
+        // Inject tracing headers into the grpc call's metadata.
+        const tracingMetadata = metadata || new grpc.Metadata()
+        const meta = {}
+
+        tracer.inject(span, FORMAT_HTTP_HEADERS, meta)
```

Comment:
The `tracingMetadata` here is an instance of the `grpc.Metadata` class. It doesn't support setting a property (you need to call `.set()` as below)

Reply:
Makes sense. I would probably extract this part to a function to make this cleaner. Also, since gRPC is not HTTP, `FORMAT_TEXT_MAP` would be more correct semantically, even do it's equivalent to `FORMAT_HTTP_HEADERS`.